### PR TITLE
Add a status and log commands to ravc2 and add tests for rvc

### PR DESCRIPTION
**Pull request by RHL120**
**Checklist**

- [ ] Closing issues: #issue
- [x] Mark this if you consider it ready to merge
- [X] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**
I have added a ravc2 log command which outputs the list of commits, a ravc2 status command which outputs thew current branch and unstaged files. I have also added tests.

### CODE REVIEW

File: libr/core/rvc.c [hunk 1]
```diff
@@ -57,6 +71,25 @@ static char *strip_sys_dir(const char *path) {
 	return ret;
 }
 
+static Sdb *vcdb_open(const char *rp) {
+	char *frp = r_str_newf ("%s" R_SYS_DIR ".rvc" R_SYS_DIR DBNAME, rp);
+	if (!frp) {
+		return NULL;
+	}
+	Sdb *ret = sdb_new0();
```
Comment (trufae) [inline libr/core/rvc.c:79]:
```suggestion
	Sdb *db = sdb_new0 ();
```

Comment (trufae) [reply libr/core/rvc.c:79]: `db` is better name than `ret`here

File: libr/core/rvc.c [hunk 2]
```diff
@@ -1003,19 +937,7 @@ R_API bool r_vc_new(const char *path) {
...
-		eprintf ("No repo in %s\nCan't checkout\n", rp);
-		return false;
-	case -1:
-		eprintf ("Can't checkout\n");
-		return false;
-	case -2:
-		eprintf ("Can't checkout");
-		return false;
-	}
+	CHECK_REPO (rp, "Can't checkout");
```
Comment (trufae) [inline libr/core/rvc.c:940]: if (!check_repo (rp, true)) {return false;}

File: test/db/tools/ravc2 [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+cd ravc2_branch_test
+!!ravc2 init
+!!ravc2 branch hello
+echo hello > hello
+!!ravc2 commit "hello world!" hello
+!!ravc2 checkout hello
+ls
+!!ravc2 status~Branch:
+!!ravc2 checkout master
+ls
```
Comment (trufae) [inline test/db/tools/ravc2:51]: you can use ls -q for one file per line which is more readable

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/core/rvc.c (hunk 2 above)
Comment (RHL120) [reply libr/core/rvc.c:940]: I think I may have misnamed repo_exists the purpose of it is not to check wheter or not a repo exists but to run digonstics on it. Do you think I should keep the functionality or change it?

File: test/db/tools/ravc2 (hunk 3 above)
Comment (RHL120) [reply test/db/tools/ravc2:51]: yes, but the reason tests are failing is because the entries are sorted differently.

**Pull request merged**